src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -245,6 +286,9 @@ void SotaUptaneClient::initialize() {
 
   verifySecondaries();
   LOG_DEBUG << "... provisioned OK";
+
+  // TODO: right place?
+  finalizeAfterReboot();
```

Comment:
Yes, I think so. The other option, particularly if we only want to incur network traffic when the user requests it, is in sendDeviceData or checkUpdates, but then we have to have logic to make sure we only send it after a reboot and not every time. This is probably good for now.